cmd/kops/validate_cluster.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+	api "k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/util/pkg/tables"
+	k8sapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+
+	"errors"
+	"fmt"
+	"os"
+)
+
+// not used too much yet :)
+type ValidateClusterCmd struct {
+}
+
+var validateClusterCmd ValidateClusterCmd
+
+// Init darn it
+func init() {
+	cmd := &cobra.Command{
+		Use:     "cluster",
+		Aliases: []string{"cluster"},
+		Short:   "Validate cluster",
+		Long:    `Validate a kubernetes cluster`,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := validateClusterCmd.Run(args)
+			if err != nil {
+				exitWithError(err)
+			}
+		},
+	}
+
+	validateCmd.cobraCommand.AddCommand(cmd)
+}
+
+// Validate Your Kubernetes Cluster
+func (c *ValidateClusterCmd) Run(args []string) error {
+
+	err := rootCommand.ProcessArgs(args)
+	if err != nil {
+		return fmt.Errorf("Process args filed %v", err)
+	}
+
+	cluster, err := rootCommand.Cluster()
+	if err != nil {
+		return fmt.Errorf("Cannot get cluster for %v", err)
+	}
+
+	clientSet, err := rootCommand.Clientset()
+	if err != nil {
+		return fmt.Errorf("Cannot get clientSet for %q: %v", cluster.Name, err)
+	}
+
+	list, err := clientSet.InstanceGroups(cluster.Name).List(k8sapi.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Cannot get nodes for %q: %v", cluster.Name, err)
+	}
+
+	fmt.Printf("Validating cluster %v\n\n", cluster.Name)
+
+	var instanceGroups []*api.InstanceGroup
+	for i := range list.Items {
+		ig := &list.Items[i]
+		instanceGroups = append(instanceGroups, ig)
+	}
+
+	if len(instanceGroups) == 0 {
+		return errors.New("No InstanceGroup objects found\n")
```

Comment:
Can we keep our error handling uniform, although `errors.New` is perfectly valid code

The rest of the project tries to use 

```
return fmt.Errorf("No InstanceGroup objects found\n")
```